### ISSUE TITLE
CI Pipeline should contain artifacts with test results (frontend/backend tests, coverage, & playwright tests)

### ISSUE THREAD

**Issue by lecegues** [UniversityOfSaskatchewanCMPT371/term-project-2024-team-4#113]
The CI Pipeline should output an artifact with test results that describe the passing and failing of tests.
Test Result Categories:

### Note
The test results are usually given as a combination of HTML, CSS, JS files that can be run as an interactive site for viewing tests & test results. Need to find a way to allow the user to "view" test results **without** having to go through much hassle

#### Frontend
- Frontend Unit/Integration Tests (Vitest)
- Frontend Coverage Tests (Vitest)

#### Backend
- Backend Unit/Integration Tests (Jest)
- Backend Coverage Tests (Jest)

#### Playwright
- E2E Tests
- Smoke Tests

**Comment by itsjorden** [comment]
Just for clarification, when you say output an "artifact" you mean a representation for the test results correct? And not a "artifact" object (as defined by the database diagram)?

**Comment by lecegues** [comment]
Yes. "Artifact" in the pretense of a CI Pipeline refers to "outputs" from the pipeline itself.

**Comment by lecegues** [comment]
Due to time constraints and the effects of leaving merging to last-minute, there are not enough resources to implement the last step of this feature (deploy to GitHub Pages). After trying to implement this, here are my observations:

### Goals & Status (refer to #114)
1. Test Scripts are created successfully. They can separately run Coverage, Unit, and Integration Tests
2. They are set up to run in the CI Pipeline and generate HTML Reports 
3. However, they cannot be deployed into GitHub actions

### Observations of Point 3. (cannot be deployed into GitHub Actions)
- The purpose of this step is to "deploy" the test reports in a viewable and in a readable manner.
1. JSON & XML are not suitable because they cannot be simply "read" by a viewer
2. Text Reporters were an option, **however**, although Jest supports text reporters to upload an artifact, Vitest does not have these reporters
3. The next choice was HTML. Jest gives **single HTML files**, which are very easy to view (click and open); **however** Vitest has an interactive website for tests, which means the HTML Directory needs to be run by a **web server**

### HTML Reporters
- Currently, the pipeline outputs `frontend/` and `backend/` artifacts with `coverage, unit, integ` HTML folders/files. The problem is that all the frontend tests need to be opened by a web server.
1. An idea was to host these HTML test documents in GitHub pages. **However**, Vitest (again) does not seem to play nice with this, paired with pair versioning issues of the GitHub actions (e.g. V4 does not play well with V2 actions), there are not enough resources to finish this.
2. **Current State**: Currently, these test documents can be viewable by running in a local web server (e.g. VSCode's LiveShare extension)
